include/pstl_offload/internal/usm_memory_replacement.h

```diff
@@ -53,34 +54,59 @@ __get_offload_device()
 class __offload_policy_holder_type
 {
     using __set_active_device_func_type = void (*)(sycl::device*);
+    using __get_offload_device_func_type = sycl::device (*)();
 
   public:
     // Since the global object of __offload_policy_holder_type is static but the constructor
     // of the class is inline, we need to avoid calling static functions inside of the constructor
     // and pass the pointer to exact function as an argument to guarantee that the correct __active_device
     // would be stored in each translation unit
-    __offload_policy_holder_type(sycl::device&& __offload_device,
+    template <typename _DeviceSelector>
+    __offload_policy_holder_type(const _DeviceSelector& __device_selector,
                                  __set_active_device_func_type __set_active_device_func)
-        : _M_offload_device(std::move(__offload_device)), _M_offload_policy(_M_offload_device),
```

Comment:
It seems like we need to also make `_M_offload_policy` std::optional. Because otherwise, the default constructor of the `device_policy` would be called that will create unnecessary `sycl::queue` with the default selector inside.

Reply:
Good point, thank you.